src/operator/subgraph/default_subgraph_op.h

```diff
@@ -121,6 +146,25 @@ class DefaultSubgraphProperty: public SubgraphProperty {
   std::shared_ptr<const std::unordered_set<std::string>> op_names_;
 };
 
+/*
+ * This subgraph property finds a subgraph whose nodes satisfy shape/types requirement.
+ * The operators in the subgraph will be executed by _default_subgraph_op.
+ */
+class ShapeTypesSubgraphProperty : public SubgraphProperty {
```

Comment:
what backend uses this subgraph property?

Reply:
it is default, similar to ContainOp selector, but i removed this for now, will submit separate PR for custom selector.